controllers/llm_controller.go

```diff
@@ -60,3 +96,110 @@ func (r *LLMReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&arcadiav1alpha1.LLM{}).
 		Complete(r)
 }
+
+// UpdateLLM updates new LLM instance.
+func (r *LLMReconciler) UpdateLLM(ctx context.Context, logger logr.Logger, instance *arcadiav1alpha1.LLM) error {
+	logger.Info("Updating LLM instance")
+	// Check if URL/Auth has changed
+	if instance.Spec.URL != instance.Status.URL || instance.Spec.Auth != instance.Status.Auth {
```

Comment:
1. URL is not empty for sure since we set it as required.
2. Here we can just test availability.

The `Auth` here is a k8s secret used to keep `APIKey`.

Reply:
Can it be decoded like below?
```go
// decode Auth into API Key
	decoded, err := base64.StdEncoding.DecodeString(instance.Spec.Auth)
	if err != nil {
		return fmt.Errorf("decode auth failed: %v", err)
	}

	auth := "Bearer " + string(decoded)
```